docs/resources/elasticsearch_component_template.md

```diff
@@ -0,0 +1,101 @@
+---
```

Comment:
All the documentation in the `docs/` folder is generated and should not be changed manually. See how it's done for `template` resource: https://github.com/elastic/terraform-provider-elasticstack/blob/0beb7d40a0f168e3e6b83ed6c579061b798e1ad4/templates/resources/elasticsearch_index_template.md.tmpl#L1-L12 

you could copy this template and change the specific parts in names and the paths to the examples and run `make gen` to generate the required files, which you then can add into this PR as well.

Reply:
I used `make docs-generate` to generate this file, but without adding the template file. I'll add it and regenete it the new documentation just after